components/HeaderUserInfo.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  Avatar,
+  Box,
+  Button,
+  HStack,
+  Menu,
+  MenuButton,
+  MenuDivider,
+  MenuItem,
+  MenuList,
+  Text,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { useLogout, useUserQuery } from "@lib/hooks/user";
+import Image from "next/image";
+import Link from "next/link";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+import LoginModal from "./LoginModal";
+
+export default function HeaderUserInfo() {
+  const { user } = useUserQuery();
+  const { isOpen, onClose, onOpen } = useDisclosure();
+
+  const [loginModalKey, setLoginModalKey] = useState("");
+  const handleLoginClick = () => {
+    // Reset login modal each time it is opened
+    setLoginModalKey(uuid());
+    onOpen();
+  };
+
+  const logout = useLogout();
+
+  if (!user) {
+    return (
+      <>
+        <Button variant="link" onClick={handleLoginClick}>
+          <HStack spacing="0">
+            <Text fontWeight="500" mr="4">
+              Login
+            </Text>
+            <Avatar size="sm" />
+          </HStack>
+        </Button>
+        <LoginModal
+          key={loginModalKey}
+          isOpen={isOpen}
+          onClose={onClose}
+          message="To sell your textbooks or contact sellers please login with your MacID below."
+        />
+      </>
+    );
+  }
+
+  return (
+    <>
+      <Menu placement="bottom-end">
+        <MenuButton aria-label="Account menu">
+          <HStack spacing="0">
+            {user.name && (
+              <Text
+                display={{ base: "none", sm: "initial" }}
+                fontWeight="500"
+                whiteSpace="nowrap"
+                mr="4"
+              >
+                {user.name}
```

Comment:
Maybe we should display the MacID if the user hasn't yet set their display name? It's kind of hard to tell when you are logged in when nothing is listed there and the profile picture is the default one.

Reply:
I like this, good idea